azext_iot/tests/iothub/devices/test_iot_edge_devices_create_int.py

```diff
@@ -275,12 +275,6 @@ def _validate_results(
         cert_auth: bool = False,
         custom_device_template: bool = False,
     ):
-        # get all devices in hub
-        device_list = self.cmd(
-            f"iot hub device-identity list -n {self.entity_name} -g {self.entity_rg}"
-        ).get_output_in_json()
-        # make sure all devices were created
-        assert len(device_list) == len(devices)
```

Comment:
thanks @c-ryan-k, you are right I deleted the length check since for existed hub, there might already be some devices so the length will not be equaled, and also I thought the code below seems like it's also checking that all devices are created successfully:
`device = self.cmd(
                f"iot hub device-identity show -d {device_id} -n {self.entity_name} -g {self.entity_rg}"
            ).get_output_in_json()` 
`assert device`

I'm thinking a way to not touch the original hub state and also able to check the length, how about save the device initial state for the hub and check here using something like `assert len(device_list) - len(original_device_list) == len(devices)`?😀

Reply:
I would also recommend looking at how the dt tests handle multiple dt instances. They keep track of all the dt instances created during a test run and in the end delete all of them

This may or may not be necessary for devices, modules, etc. (Think about what happens if you let the test run fully, cancel mid test, etc)